yarn-project/simulator/src/public/executor.ts

```diff
@@ -40,29 +34,20 @@ export class PublicExecutor {
   static readonly log = createDebugLogger('aztec:simulator:public_executor');
 
   /**
-   * Executes a public execution request.
+   * Simulate a public execution request.
    * @param executionRequest - The execution to run.
-   * @param constants - The constants (including global variables) to use.
+   * @param globalVariables - The global variables to use.
    * @param allocatedGas - The gas available at the start of this enqueued call.
-   * @param txContext - Transaction context.
-   * @param pendingSiloedNullifiers - The pending nullifier set from earlier parts of this TX.
    * @param transactionFee - Fee offered for this TX.
-   * @param startSideEffectCounter - The counter of the first side-effect generated by this simulation.
-   * @param previousValidationRequestArrayLengths - Side effect array lengths from previous kernel
-   * @param previousAccumulatedDataArrayLengths - Side effect array lengths from previous kernel
-   * @returns The result of execution, including the results of all nested calls.
+   * @returns The result of execution.
    */
   public async simulate(
-    executionRequest: PublicExecutionRequest,
-    constants: CombinedConstantData,
+    stateManager: AvmPersistableStateManager,
+    executionRequest: PublicExecutionRequest, // TODO(dbanks12): CallRequest instead?
```

Comment:
CallRequest only contains args_hash, that's why we use this PublicExecutionRequest in ts to pass the actual args.